### PR TITLE
Updating CallInfo to be more in line with Context API.

### PR DESCRIPTION
**Pull request by alainjobart**
And remove all the cast madness.
@sougou @enisoc

### CODE REVIEW

File: go/vt/tabletserver/streamlogger.go [hunk 1]
```diff
@@ -150,50 +154,47 @@ func (stats *SQLQueryStats) FmtQuerySources() string {
...
-		log.WaitingForConnection.Seconds(),
-		log.RowsAffected,
-		log.SizeOfResponse(),
-		log.CacheHits,
-		log.CacheMisses,
-		log.CacheAbsent,
-		log.CacheInvalidations,
-		log.ErrorStr(),
+		stats.Method,
+		remoteAddr,
```
Comment (sougou) [inline go/vt/tabletserver/streamlogger.go]: This may break some backward compatibility. I checked that querylogz.go doesn't use this, but others may.

File: go/vt/tabletmanager/gorpctmserver/gorpc_server.go [hunk 2]
```diff
@@ -35,6 +36,7 @@ type TabletManager struct {
 
 // Ping wraps RPCAgent.
 func (tm *TabletManager) Ping(ctx context.Context, args, reply *string) error {
+	ctx = callinfo.RPCWrapCallInfo(ctx)
```
Comment (sougou) [inline go/vt/tabletmanager/gorpctmserver/gorpc_server.go]: Why is this needed only for TabletManager and not for others like SqlQuery, VTGate, etc?

Comment (alainjobart) [inline go/vt/tabletmanager/gorpctmserver/gorpc_server.go]: It is needed if the server uses the CallInfo extra data (otherwise, no need to add it, it's just extra work for nothing).

The only two that do for now are:
- sqlquery
- tmserver
I think I covered both in this CL.

File: go/vt/tabletserver/streamlogger.go (hunk 1 above)
Comment (alainjobart) [inline go/vt/tabletserver/streamlogger.go]: Why? there is no change here, really. RemoteAddr() and Username would just return "" if there was no context before. So it's just the exact same result as before... Or am I missing something?

File: go/vt/tabletmanager/gorpctmserver/gorpc_server.go (hunk 2 above)
Comment (alainjobart) [inline go/vt/tabletmanager/gorpctmserver/gorpc_server.go]: I merged this, so I can run the import and fix all the associated files
this morning... I'll fix forward if I missed something.
